plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/DiagramObjectCollector.java

```diff
@@ -213,11 +211,8 @@ private boolean aliasExist(String alias) {
                 }
             }
             final DBPProject project = ((DBSObject) object).getDataSource().getContainer().getProject();
-            if (currentProject == null) {
-                currentProject = project;
-            }
-            if (currentProject != project) {
-                diagram.addErrorMessage("Can't add object '" + DBUtils.getObjectFullName(object, DBPEvaluationContext.UI) + "' from different project '" + project.getName() + "' (current project is '" + currentProject.getName() + "')");
+            if (activeProject != project) {
+                diagram.addErrorMessage("Can't add object '" + DBUtils.getObjectFullName(object, DBPEvaluationContext.UI) + "' from a non-active project '" + project.getName() + "' (active project is '" + activeProject.getName() + "')");
```

Comment:
If I understand this correctly, if the user selects multiple objects from a non-active project and tries to add them to a diagram, we will show them a bunch of consequent identical error messages. That doesn't look good.

Reply:
Improved:

![java_T1KZzbUIOR](https://user-images.githubusercontent.com/35821147/129006533-af5a71d3-17e2-403e-ae56-56e367512699.png)